src/fivetran-destination/src/destination/dml.rs

```diff
@@ -388,6 +407,152 @@ async fn update_file(
+
+    // Create our temporary table that we'll copy the delete files into.
+    let defs = columns
+        .iter()
+        .map(|col| format!("{} {}", col.name, col.ty))
+        .join(",");
+    let create_table_stmt = format!("CREATE TABLE {schema_name}.{temp_table_name} ({defs})");
+    client.batch_execute(&create_table_stmt).await?;
+
+    // TODO(parkmycar): Ideally we'd use the COPY FROM protocol here, but we need to add support
+    // for compression and encryption.
+    let cols = columns
+        .iter()
+        .enumerate()
+        .map(|(idx, col)| format!("${}::{}", idx + 1, col.ty))
+        .join(",");
+    let insert_stmt = format!("INSERT INTO {schema_name}.{temp_table_name} VALUES ({cols})",);
+    let insert_stmt = client
+        .prepare(&insert_stmt)
+        .await
+        .context("internal error: preparing delete temp table insert statement")?;
+
+    let mut total_count = 0;
+    for path in delete_files {
+        let reader = load_file(file_config, path)
+            .await
+            .with_context(|| format!("loading delete file {path}"))?;
+        let copied = copy_delete_file(file_config, client, &insert_stmt, reader)
+            .await
+            .with_context(|| format!("handling delete file {path}"))?;
+        total_count += copied;
+    }
+    tracing::info!(total_count, "inserted rows to {schema}.{temp_table_name}");
+
+    // Skip the update if there are no rows to delete!
+    if total_count > 0 {
+        // Mark rows as deleted by "merging" our temporary table into the destination table.
+        //
+        // TODO(parkmycar): Fivetran provides a column name in truncate for soft deletes, should the
+        // delete operation be doing the same?
+
+        // HACKY: We want to update the "_fivetran_synced" column for all of the rows we marked as
+        // deleted, but don't have a way to read from the temp table that would allow this in an
+        // `UPDATE` statement.
+        let synced_time_stmt = format!(
+            "SELECT MAX({synced_col}) FROM {schema_name}.{temp_table_name}",
+            synced_col = escape::escape_identifier(FIVETRAN_SYSTEM_COLUMN_SYNCED)
+        );
+        let synced_time: SystemTime = client
+            .query_one(&synced_time_stmt, &[])
+            .await
+            .and_then(|row| row.try_get(0))?;
+
+        let matching_cols = columns.iter().filter(|col| col.is_primary);
+        let merge_stmt = format!(
+            r#"
+            UPDATE {schema_name}.{table_name}
+            SET {deleted_col} = true, {synced_col} = $1
+            WHERE ({cols}) IN (
+                SELECT {cols}
+                FROM {schema_name}.{temp_table_name}
+            )"#,
+            deleted_col = escape::escape_identifier(FIVETRAN_SYSTEM_COLUMN_DELETE),
+            synced_col = escape::escape_identifier(FIVETRAN_SYSTEM_COLUMN_SYNCED),
+            cols = matching_cols.map(|col| &col.name).join(","),
+        );
```

Comment:
What I'm trying to do here is update the value of `_fivetran_synced` for all of the rows we marked as deleted, based on the value provided in the temp table. It doesn't seem like that's possible to do in an `UPDATE` statement, without using a `FROM` clause, which I don't think we support today?